clientconn.go

```diff
@@ -213,10 +216,12 @@ func WithUserAgent(s string) DialOption {
 
 // Dial creates a client connection the given target.
 func Dial(target string, opts ...DialOption) (*ClientConn, error) {
+	ctx := context.Background()
 	cc := &ClientConn{
 		target: target,
 		conns:  make(map[Address]*addrConn),
 	}
+	cc.ctx, cc.cancel = context.WithCancel(ctx)
```

Comment:
I'd prefer to keep it. The next step here is to add a DialContext method, which will accept a user-provided context; then Dial becomes a thin wrapper; leaving line 219 will reduce the overall diff.